warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ValidPatternVisitor.java

```diff
@@ -69,22 +56,73 @@ public Object visit(ASTERNode node, Object data) {
      */
     @Override
     public Object visit(ASTNRNode node, Object data) {
+        parseAndPutLiteral(node);
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTFunctionNode node, Object data) {
+        
+        if (node.jjtGetNumChildren() >= 4) {
```

Comment:
I will have to say that a more robust solution here is to push this functionality into the function descriptor mechanism.  That way we can catch any function that might have a regex......

Reply:
The JexlArgumentDescriptor already has a regexArguments() method to determine if the literal arguments are to be treated as regexes.  We should use that along with fieldsForNormalization to determine which arguments to test.

Reply:
And then we can apply this to all functions that return true for regexArguments().....

Reply:
I took a look at some of the existing descriptors and see where you're going with this. Should have an update for this pr tomorow sometime

Reply:
An example of how to do this can be found in the datawave.query.jexl.visitors.FunctionNormalizationRebuildingVisitor.